File: python/leetcode/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max = 0
        min = None
        biggest = 0
        for p in prices:
            if min is None or min > p:
                min = p
            if (p - min) > biggest:
                biggest = p - min
        return biggest

File: ruby/leetcode/best-time-to-buy-and-sell-stock.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    max = 0
    min = nil
    biggest = 0
    prices.each do |p|
        if min == nil || min > p
            min = p
        end
        if (p - min) > biggest
            biggest = p - min
        end
    end
    return biggest
end
